test/server/cards/02_SHD/units/KihraxzHeavyFighter.spec.ts

```diff
@@ -0,0 +1,38 @@
+describe('Kihraxz Heavy Fighter', function () {
+    integration(function (contextRef) {
+        it('should gain +3 on attack if you exhaust another friendly unit and self is not selectable', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    groundArena: ['battlefield-marine', 'pyke-sentinel'],
+                    spaceArena: ['kihraxz-heavy-fighter']
+                },
+                player2: {
+                    spaceArena: ['devastator#inescapable']
+                }
+            });
+            const { context } = contextRef;
+
+            context.player1.clickCard(context.kihraxzHeavyFighter);
+            expect(context.player1).toBeAbleToSelectExactly(['battlefield-marine', 'pyke-sentinel']);
+            context.player1.clickCard(context.battlefieldMarine);
+            expect(context.devastatorInescapable.damage).toBe(6);
+        });
+
+        it('should do normal attack if no unit can be exhausted', () => {
```

Comment:
This test could be merged into the above by setting exhausted = false. See Rukh tests for a good example of how we set that up (we have to combine test cases because we need to be careful about our usage of Github Actions time).

Combining them also gives us the advantage of confirming that the +3 goes away after the first attack as expected.